Makefile

```diff
@@ -163,8 +163,15 @@ prep: check-go-version
 	@GOARCH= GOOS= $(GO_CMD) generate $(MAIN_PACKAGES)
 	@GOARCH= GOOS= cd api && $(GO_CMD) generate $(API_PACKAGES)
 	@GOARCH= GOOS= cd sdk && $(GO_CMD) generate $(SDK_PACKAGES)
+
+# Git doesn't allow us to store shared hooks in .git. Instead, we make sure they're up-to-date
+# whenever a make target is invoked.
+.PHONY: hooks
+hooks:
 	@if [ -d .git/hooks ]; then cp .hooks/* .git/hooks/; fi
 
+-include hooks # Make sure they're always up-to-date
```

Comment:
My make-fu has never been very strong, could you explain what this does please?  I read that "-include filename" means don't error if filename doesn't exist, but here it doesn't seem that there's a file involved?

Reply:
When Make is reading the Makefile and it comes across `-include` it suspends and executes whatever is there before continuing. In our case we're using a PHONY target instead of a file so it always executes the target before continuing. That has the result of always making sure our hooks are up-to-date before processing any other called targets.

Reply:
Thanks!